src/System.ServiceModel.Primitives/tests/Description/OperationBehaviorTest.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ServiceModel;
+using Xunit;
+
+public static class OperationBehaviorTest
+{
+    [Fact]
+    [ActiveIssue(315)]
+    public static void IOperationBehavior_Methods_AreCalled()
+    {
+        DuplexClientBase<ICustomOperationBehaviorDuplexService> duplexService = null;
+        ICustomOperationBehaviorDuplexService proxy = null;
+
+        NetTcpBinding binding = new NetTcpBinding();
+        binding.Security.Mode = SecurityMode.None;
+
+        WcfDuplexServiceCallback callbackService = new WcfDuplexServiceCallback();
+        InstanceContext context = new InstanceContext(callbackService);
+
+        duplexService = new MyDuplexClientBase<ICustomOperationBehaviorDuplexService>(context, binding, new EndpointAddress(FakeAddress.TcpAddress));
+        proxy = duplexService.ChannelFactory.CreateChannel();
+
+        // Wait to validate until the process has been given a reasonable time to complete.
+        MyOperationBehavior.mre.WaitOne(250);
+
```

Comment:
We should add a test about WaitOne's return value.  If it returns 'true' the mre.Set was called.  If it return 'false' it means it timed out.  So we should Assert.True(waitOneResult, "IOperationBehavior methods were not called").  It might also be useful that in the 'false' case it shows which of the methods *were* called.